google/cloud/bigquery/v2/minimal/internal/job_rest_connection_impl.cc

```diff
@@ -39,6 +39,17 @@ StatusOr<GetJobResponse> BigQueryJobRestConnectionImpl::GetJob(
       request, __func__);
 }
 
+StatusOr<ListJobsResponse> BigQueryJobRestConnectionImpl::ListJobs(
+    ListJobsRequest const& request) {
+  return rest_internal::RestRetryLoop(
```

Comment:
We have a helper to do the pagination logic. The implementation would look more like the following:

https://github.com/googleapis/google-cloud-cpp/blob/c0e30ed9c067609411bdbed5e0440be4cb8e653c/generator/integration_tests/golden/v1/internal/golden_kitchen_sink_rest_connection_impl.cc#L79-L104